internal/mcp-router/request_handlers.go

```diff
@@ -0,0 +1,174 @@
+package mcprouter
```

Comment:
The removal of request.go is actually moving existing logic into request_handlers.go, or is there more to it than that?

Reply:
It moves the existing logic and pulls out the ext_proc specific responses into the `ResponseBuilder` type. Hopefully making it all a bit more readable extendible and maintainable